testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/RunAsPrincipalTestCase.java

```diff
@@ -218,4 +232,29 @@ private Throwable checkEjbException(Throwable ex) {
         }
         return ex;
     }
+
+    private boolean logContainsWARN() {
+        String jbossHome = System.getProperty("jboss.inst");
+        String logPath = String.format("%s%sstandalone%slog%sserver.log", jbossHome,
+                (jbossHome.endsWith(File.separator) || jbossHome.endsWith("/")) ? "" : File.separator,
+                File.separator, File.separator);
+        logPath = logPath.replace('/', File.separatorChar);
+        Stream<String> fileStream = null;
+        try {
+            fileStream = Files.lines(Paths.get(logPath)); // UTF8 is used by default
+        } catch (MalformedInputException e1) {
+            // some windows machines could accept only StandardCharsets.ISO_8859_1 encoding
+            try {
+                fileStream = Files.lines(Paths.get(logPath), StandardCharsets.ISO_8859_1);
+            } catch (IOException e4) {
+                throw new RuntimeException("Server logs has not standard Charsets (UTF8 or ISO_8859_1)");
+            }
+        } catch (IOException e) {
+            // server.log file is not created, it is the same as server.log is empty
+        }
+        if(fileStream != null) {
+            return fileStream.filter(lines -> lines.contains("@RunAs annotation is required when using @RunAsPrincipal")).iterator().hasNext();
```

Comment:
We're starting to have a lot of this log searching. For stuff where we expect something logged, perhaps use something like https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/input/ReversedLinesFileReader.html to search from the end rather than starting many thousands of lines before?

@jamezp WDYT?

Reply:
@bstansberry In this case, the reverse lines I think is not useful since I expect the log line on EJB application deployment time, at the beginning. I was wondering if you already have something in place that can be reused to search for something in the logs.

Reply:
@tmiyargi The server.log covers the entire surefire execution which in the case of testsuite/integration/basic might mean logging from hundreds of tests and dozens of reloads.

For example, in the server log from one of the CI jobs for this PR, the WARN is logged on line number 20481.

https://ci.wildfly.org/repository/download/WF_PullRequest_LinuxJdk11/155885:id/testsuite/integration/basic/target/wildfly/standalone/log/server.log

I don't know if there are any utilities around this kind of thing. @jamezp might know.

One idea I just had is a server setup thing that adds a new test-class-specific logging handler and then removes it at the end. So the scan can just cover the file from that new handler. That would work for both this kind of case, where you are asserting the presence of something, and for cases where the assertion is something isn't logged. (I particularly hate those kinds of tests, but people write them, since bug fixes must have regression tests. I'd hate those tests less if they were'n't scanning so much.)

Reply:
There is no utility that I'm aware of either, but I do like the idea of having some kind of common handler that is added for tests like this. I don't really like tests checking for log messages however sometimes it's the only option. I'd be happy to look at a utility for this if interested. A benefit of a common `ServerSetupTask` is it could take something like the log level and the log category to reduce the I/O.

One thing I find is useful is to use the `json-formatter` so we have some structured data instead. Though I guess if we're just looking for message id's it might not be a big deal.

Reply:
So, I created the utility class, let me know if it's fine. Thank you Brian and James.

Reply:
Thanks @tmiyargi!